### ISSUE TITLE
`finalize` method should be private

### ISSUE THREAD

**Issue by wch** [r-lib/R6#146]
I'm not sure why I originally made it so the `finalize` method was in `public`. It makes more sense for it to be in `private`.

**Comment by gaborcsardi** [comment]
True, but this is a breaking change, right? Or can you implement sg. that helps the transition? E.g. automatically moving it over to private, with a warning?

**Comment by wch** [comment]
Yes, it would be breaking, so it would be a good idea to do something like what you propose. Maybe start with messages and then in, say, a year, upgrade it to a warning.

**Comment by wch** [comment]
I just released v2.4.0, which allows private finalizers. The documentation now says that finalizers should be private, but they are allowed to be public or private.

In the future -- maybe in a year or so -- we can make R6 start emitting messages or warnings for public finalizers, and then some time after that, we can stop supporting public finalizers. I'm reopening this issue as a reminder.

**Comment by llrs-roche** [comment]
I'm trying to estimate how close we are to getting warning for public finalizers (and thus potential CRAN notices) and I'm a bit puzzled by the dates.  Was this released on 2019 as the previous comments suggested or recently in 2025?  The changelog and the R version where this item appears was released on 2025 on R6 2.6.0 instead of on v2.4.0. If it is the former I guess dependencies still have 1 year to move to private finalizers. Thanks!
